### PR TITLE
[PKX-009] [Enhancement] Use Loading widget

### PR DESCRIPTION
**Pull request by giantiyan**

### CODE REVIEW

File: lib/features/about_tab/about_tab_vm.dart [hunk 1]
```diff
@@ -1,15 +1,34 @@
...
+      return const UnionPageState.error('About Tab Error Message');
+    }
+  }
 }
 
 class AboutTabVm extends Vm {
-  AboutTabVm({this.about}) : super(equals: [about]);
+  AboutTabVm({
+    this.about,
+    required this.pageState,
```
Comment (SaturnoMark) [inline lib/features/about_tab/about_tab_vm.dart:28]: please move at top

File: lib/features/base_stats_tab/base_stats_tab_vm.dart [hunk 2]
```diff
@@ -1,15 +1,34 @@
...
+      return const UnionPageState.error('Base Stats Tab Error Message');
+    }
+  }
 }
 
 class BaseStatsTabVm extends Vm {
-  BaseStatsTabVm({this.baseStats}) : super(equals: [baseStats]);
+  BaseStatsTabVm({
+    this.baseStats,
+    required this.pageState,
```
Comment (SaturnoMark) [inline lib/features/base_stats_tab/base_stats_tab_vm.dart:28]: same above

File: lib/features/details_page/details_page_vm.dart [hunk 3]
```diff
@@ -9,15 +11,28 @@ class DetailsPageVmFactory extends VmFactory<AppState, DetailsPageConnector> {
...
+    }
+  }
 }
 
 class DetailsPageVm extends Vm {
   DetailsPageVm({
     this.specificPokemon,
     this.types,
-  }) : super(equals: [specificPokemon, types]);
+    required this.pageState,
```
Comment (SaturnoMark) [inline lib/features/details_page/details_page_vm.dart]: same above

File: lib/features/evolution_tab/evolution_tab_vm.dart [hunk 4]
```diff
@@ -1,15 +1,34 @@
...
+      return const UnionPageState.error('Evolution Tab Error Message');
+    }
+  }
 }
 
 class EvolutionTabVm extends Vm {
-  EvolutionTabVm({this.evolutions}) : super(equals: [evolutions]);
+  EvolutionTabVm({
+    this.evolutions,
+    required this.pageState,
```
Comment (SaturnoMark) [inline lib/features/evolution_tab/evolution_tab_vm.dart:28]: same above

File: lib/features/moves_tab/moves_tab_vm.dart [hunk 5]
```diff
@@ -1,15 +1,34 @@
...
+      return const UnionPageState.error('Moves Tab Error Message');
+    }
+  }
 }
 
 class MovesTabVm extends Vm {
-  MovesTabVm({this.moves}) : super(equals: [moves]);
+  MovesTabVm({
+    this.moves,
+    required this.pageState,
```
Comment (SaturnoMark) [inline lib/features/moves_tab/moves_tab_vm.dart:28]: same above

File: lib/features/home_page/home_page_vm.dart [hunk 6]
```diff
@@ -1,15 +1,34 @@
...
+      return const UnionPageState.error('Home Page Error Message');
+    }
+  }
 }
 
 class HomePageVm extends Vm {
-  HomePageVm({this.pokemon}) : super(equals: [pokemon]);
+  HomePageVm({
+    this.pokemon,
+    required this.pageState,
```
Comment (SaturnoMark) [inline lib/features/home_page/home_page_vm.dart:28]: same above

**Review by SaturnoMark — CHANGES_REQUESTED** [review]

**Review by SaturnoMark — APPROVED** [review]

**Review by ryle-roasa — APPROVED** [review]

**Pull request merged**